### PR TITLE
adding more instruments to llama_index agents

### PR DESCRIPTION
**Pull request by piotrm0**
Instrumenting more components of the llama_index yelp agent example. Timeline now shows more details (see screenshots). Added component categories to the dashboard for Agent, Tool, and added LLM component for llama_index LLMs.

Screenshot 2023-09-11 at 18 03 14

# Add to Release notes:

Change to layout of calls in `Record.layout_calls_as_app` affects how selectors are written. Assume instrumented app has a method `somemethod` which is called more than once on an invocation of the app. Previously, `Select.Record.app.somemethod` would contain call information for only the last call of that method. Now it will contain all calls in a sequence. To get the ith, you can use index notation: `Select.Record.app.somemethod[i]`. However, to make this change somewhat backwards compatible, if there is only one call, the index can be omitted. That is, `Select.Record.app.somemethod[0].rets...` is equivalent to `Select.Record.app.somemethod.rets...`. A warning will be given if there is more than one call and an index is not used to disambiguate them.

### CODE REVIEW

**Review by joshreini1 — CHANGES_REQUESTED** [review]
Can you update the key setting method to the more explicit setting? This way it’s more easily runnable in Google colab.

File: trulens_eval/trulens_eval/app.py [hunk 1]
```diff
@@ -588,19 +607,19 @@ def _get_method_path(self, obj: object, func: Callable) -> JSONPath:
...
+            logger.warn(
                 f"A new object of type {type(obj)} at 0x{id(obj):x} is calling an instrumented method {func}. "
                 "The path of this call may be incorrect."
             )
             try:
                 _id, f, path = next(iter(self._get_methods_for_func(func)))
             except Exception:
-                logger.debug(
+                logger.warn(
                     "No other objects use this function so cannot guess path."
```
Comment (joshreini1) [inline trulens_eval/trulens_eval/app.py:618]: Is guessing the best language to use here?

Comment (piotrm0) [reply trulens_eval/trulens_eval/app.py:618]: I think so, this point of the code is meant to guess a path (pick one without being sure it is correct).

**Comment by piotrm0** [comment]
> Can you update the key setting method to the more explicit setting? This way it’s more easily runnable in Google colab.

I think I fixed. Let me know.

**Review by joshreini1 — APPROVED** [review]

**Pull request merged**